### PR TITLE
`2.0.0` - FortCanning Upgrade

### LINKED ISSUE #761: `2.0.0` - FortCanning Upgrade Regression
#### What would you like to be added:

Check if there are any test regressions in this update. An early PR is created at https://github.com/DeFiCh/jellyfish/pull/753

### PR DESCRIPTION
**Pull request by fuxingloh**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #761

### CODE REVIEW

**Comment by monstrobishi** [comment]
For

```
FAIL packages/jellyfish-api-core/__tests__/category/poolpair/listPoolPairs.test.ts (53.007 s)
  ● Poolpair › should listPoolPairs

    expect(received).toStrictEqual(expected) // deep equality

    Expected: 3
    Received: 2
```

The only thing I can find was this change from 2 weeks ago:

https://github.com/DeFiCh/ain/blob/cdf7e7539d4e095cfe5127a998c11541f807f0bc/src/masternodes/rpc_poolpair.cpp#L193

I'm not sure if it's related, but it could be.

**Comment by ivan-zynesis** [comment]
> A bunch of the tests e.g. account tests are giving:
> 
> ```
> waitForWalletBalanceGTE is not ready within given timeout of 30000ms
> ```
> 
> Likely just flakey setup or need to increase the timeout

`waitForWalletBalanceGTE` is basically waiting more blocks, mostly happened on ICX related test, the consensus logic getting heavy after ICX introduced, 30 secs no longer enough for that much of blocks mined.

**Comment by ivan-zynesis** [comment]
One of the issue is deprecated `getGov`, data structure changed.

**Comment by fuxingloh** [comment]
> One of the issue is deprecated `getGov`, data structure changed.

Damn... we can't replace `getGov` with another RPC seems like we have replace the original function due to data structure change.

https://github.com/DeFiCh/jellyfish/pull/794/files#diff-4d7e75f85e53bcd52f372263dedc6e010c564ed07bf046faf3f5024704a08ac3R166

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_dex_poolswap.test.ts [hunk 1]
```diff
@@ -160,75 +161,44 @@ describe('dex.poolswap()', () => {
     expect(prevouts[0].value.toNumber()).toBeGreaterThan(9.999)
   })
 
-  it('should fail poolSwap due to maxPrice lower than reserveA/reserveB', async () => {
```
Comment (ivan-zynesis) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_dex_poolswap.test.ts:163]: Duplicated test, `reserveA/reserveB` is the price.

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_dex_poolswap.test.ts [hunk 2]
```diff
@@ -160,75 +161,44 @@ describe('dex.poolswap()', () => {
...
-    await sendTokensToAddress(container, await providers.getAddress(), 2, 'BIRD')
+    await sendTokensToAddress(container, await providers.getAddress(), 2, 'GOAT')
     await fundEllipticPair(container, providers.ellipticPair, 10)
     const script = await providers.elliptic.script()
 
     const txn = await builder.dex.poolSwap({
       fromScript: script,
-      fromTokenId: pairs.BIRD.tokenA,
-      fromAmount: new BigNumber('2'),
+      fromTokenId: pairs.GOAT.tokenB,
```
Comment (ivan-zynesis) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_dex_poolswap.test.ts:183]: Change multiple price related tests from `convert DFI to other` to `convert other to DFI`, minting other token is way faster than wait for DFI token balance.

Large token volume reduce the slippage effect and improve the test accuracy

**Review by ivan-zynesis — APPROVED** [review]

**Review by jingyi2811 — DISMISSED** [review]

**Review by canonbrother — APPROVED** [review]

File: packages/jellyfish-api-core/__tests__/category/masternode/getGov.test.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ describe('Masternode', () => {
 
   it('should getGov LP_DAILY_DFI_REWARD', async () => {
     const gov = await client.masternode.getGov('LP_DAILY_DFI_REWARD')
-    expect(gov.LP_DAILY_DFI_REWARD instanceof BigNumber).toStrictEqual(true)
+    expect(gov[0].LP_DAILY_DFI_REWARD instanceof BigNumber).toStrictEqual(true)
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/masternode/getGov.test.ts:22]: are we reverting `getgov` array changes?

Comment (fuxingloh) [reply packages/jellyfish-api-core/__tests__/category/masternode/getGov.test.ts]: reverted

**Review by jingyi2811 — APPROVED** [review]

**Review by ivan-zynesis — APPROVED** [review]

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_set_oracle_data.test.ts [hunk 4]
```diff
@@ -136,7 +137,7 @@ describe('set oracle data', () => {
     // Set Oracle Data
     const setDataTxn = await newBuilder.oracles.setOracleData({
       oracleId: oracleId,
-      timestamp: new BigNumber('1621567932'),
+      timestamp: new BigNumber(Math.floor(new Date().getTime() / 1000)),
```
Comment (monstrobishi) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_set_oracle_data.test.ts:140]:
```suggestion
      timestamp: new BigNumber(Math.floor(Date.now() / 1000)),
```

Comment (fuxingloh) [reply packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_set_oracle_data.test.ts:140]: Will move this fix to next PR

**Pull request merged**